cpp/coord/tests.cpp

```diff
@@ -0,0 +1,874 @@
+	
+	stage += 1; // 1
+	// Test camhud to window
+	
+	// camhud is always the same x as window but the height of the
+	// screen below window
+	expectedW = {0, 600};
+	c = {0, 0};
+
+	resultsW = c.to_window();
+	if( not(resultsW == expectedW)) { return stage; }
+
+	stage += 1; // 2
+	// Test camhud to term
+	
+	// TODO The expected results for this was created by just seeing
+	// What the results acually were. Need to investigate the bottomleft
+	// and top right of console as a toptight that is the camhud type
+	// is negative in its y and thus would be off the screen
+
+	// Since we are not running from the main engine, get the cwd
+	char currentPath[128];
+	char* stop;
+	getcwd(currentPath, sizeof(currentPath));
+
+	// Cut off the un-needed stuff
+	stop = strstr(currentPath, ".bin");
+	stop--;
+	*stop = 0;
+
+	// Add the correct assets dir
+	util::Dir assetDir(currentPath);
+	assetDir = assetDir.append("assets/converted");
+
+	auto termcolors = util::read_csv_file<gamedata::palette_color>(assetDir.join("termcolors.docx"));
+	console::Console con(termcolors);
+
+	window size{80*con.charsize.x, 25*con.charsize.y};
+	con.on_resize(size);
+
+	stage += 1; //  3
+
+	term expectedT{1, -54};
+	c = {12, 6};
+	
+	term resultsT = c.to_term(&con);
+	if( not(resultsT == expectedT)) { return stage; }
+
+	return -1;
+}
+
+/**
+ * This function tests the methods of camhud_delta.
+ */
+int camhud_delta_0() {
+	int stage = 0;
+	// Test camhud_delta to window_delta, show y is flipped
+	// TODO The expected results for this was created by just seeing
+	// What the results acually were. Need to investigate the bottomleft
+	// and top right of console as a toptight that is the camhud type
+	// is negative in its y and thus would be off the screen
+
+	window_delta expectedWD{100, -100};
+	camhud_delta cD{100, 100};
+
+	window_delta resultsWD = cD.to_window();
+	if( not(resultsWD == expectedWD)) { return stage; }
+	
+
+	return -1;
+}
+
+/**
+ * This function tests the methods of term
+ */
+int term_0() {
+	int stage = 0;
+
+	// Since we are not running from the main engine, get the cwd
+	char currentPath[128];
+	char* stop;
+	getcwd(currentPath, sizeof(currentPath));
+
+	// Cut off the un-needed stuff
+	stop = strstr(currentPath, ".bin");
+	stop--;
+	*stop = 0;
+
+	// Add the correct assets dir
+	util::Dir assetDir(currentPath);
+	assetDir = assetDir.append("assets/converted");
+
+	auto termcolors = util::read_csv_file<gamedata::palette_color>(assetDir.join("termcolors.docx"));
+	console::Console con(termcolors);
+
+	stage += 1; // 1
+
+	camhud expectedC{40, -84};
+	term t{5, 5};
+
+	camhud resultsC = t.to_camhud(&con);
+	if( not(resultsC == expectedC)) { return stage; }
+
+	return -1;
+}
+
+/**
+ * Top level function for coord testing.
+ */
+void coord() {
+	int ret;
+	const char *testname;
+
+	// Some tests below need the engine to be initialized
+	util::Dir dataDir{"assets"};
+	Engine::create( &dataDir ,"testing openage");
```

Comment:
Yeah I kind of figured that once I saw the build failures on travis. I'll have a look into how we can get around it.